src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py

```diff
@@ -0,0 +1,696 @@
+            # interpolated in the UI
+            else:
+                object_inital_state = key_ann["annotationValue"]["vertices"]
+                object_final_state = next_key_ann["annotationValue"]["vertices"]
+                norm_vertices = _interpolate_object(
+                    object_type=json_interface["jobs"][annotation["job"]]["tools"][0],
+                    object_initial_state=object_inital_state,
+                    initial_state_frame_index=key_ann_start,
+                    object_final_state=object_final_state,
+                    final_state_frame_index=key_ann_end,
+                    at_frame=frame_id,
+                )
+
+            if json_interface["jobs"][annotation["job"]]["tools"][0] == "marker":
+                annotation_dict["point"] = norm_vertices[0][0][0]
+
+            elif json_interface["jobs"][annotation["job"]]["tools"][0] in {"polygon", "rectangle"}:
+                annotation_dict["boundingPoly"] = [{"normalizedVertices": norm_vertices[0][0]}]
+
+            elif json_interface["jobs"][annotation["job"]]["tools"][0] == "semantic":
+                annotation_dict["boundingPoly"] = [
+                    {"normalizedVertices": norm_vert} for norm_vert in norm_vertices[0]
+                ]
+
+            json_resp[str(frame_id)].setdefault(annotation["job"], {}).setdefault(
+                "annotations", []
+            ).append(annotation_dict)
+
+    return json_resp
+
+
+def _interpolate_object(
+    *,
+    object_type: str,
+    object_initial_state: List[List[List[Vertice]]],
+    initial_state_frame_index: int,
+    object_final_state: List[List[List[Vertice]]],
+    final_state_frame_index: int,
+    at_frame: int,
+) -> List[List[List[Vertice]]]:
+    """Interpolate an object between two key frames."""
+    # if the two frames are consecutive, we do not interpolate
+    if at_frame == initial_state_frame_index:
+        return object_initial_state
+
+    if at_frame == final_state_frame_index:
+        return object_final_state
+
+    if object_type == JobTool.MARKER:
+        # for point jobs, we interpolate for each frame
+        return [
+            [
+                [
+                    _interpolate_point(
+                        previous_point=object_initial_state[0][0][0],
+                        next_point=object_final_state[0][0][0],
+                        weight=(at_frame - initial_state_frame_index)
+                        / (final_state_frame_index - initial_state_frame_index),
+                    )
+                ]
+            ]
+        ]
+
+    if object_type == JobTool.RECTANGLE:
+        return [
+            [
+                _interpolate_rectangle(
+                    previous_vertices=object_initial_state[0][0],
+                    next_vertices=object_final_state[0][0],
+                    width=1,
+                    height=1,
+                    weight=(at_frame - initial_state_frame_index)
```

Comment:
@baptiste-olivier I'm not sure about setting this to 1. Do we really need to fetch the assets' resolutions too for each label?